docs/architecture/architecture.tex

```diff
@@ -34,69 +34,97 @@
-		\end{itemize}
-
-\section{Software architecture views}
-\subsection{Subsystem decomposition}
-
-\begin{figure}[h]
-	\centering
-   	\includegraphics[scale=0.6]{images/modules.jpg}
-    \caption{Modules of the system}
-    \label{fig:modules}
-\end{figure}
-The input module should be able to read the data. With that data a dataModel can be constructed. \\
-The Analysis Specification module processes the script that defines how the data should be analyzed. The user provides this script. This module translate the script in operations that can be performed by the Analysis Module.\\
-The Analysis Module perform the analyses over the dataModel. The output of this module is the result of the analysis, this is a dataModel.\\
-The Visualization module creates a certain visual representation of a dataModel. For example it can create a box-plot of the creatine levels.
-
-
-\section{Software architecture}
-
-\subsection{Programming languages and programs}
-
-\subsection{Used librarians}
-
+This document describes the design of the architecture and it is continuously updated to the current state of the software. It also states the design goals we want to achieve during the project. Furthermore the subsystem decomposition is laid out.
 
 \section{Design goals}
+\subsection{Modularity}
+The program can be split into a number of modules. These modules should be implemented independently of the other modules. Interaction between modules is only possible by making use of defined interfaces.
 
-\subsection{Availability}
-
-\subsection{Manageability}
+\subsection{Continuous Integration}
+There should be always a working product. Each sprint should add one or more feathers.
 
-\subsection{Performance}
+\subsection{Understandability}
+We must also make the software user-friendly. Even though the users of the
+product are already familiar with analysis software, our application should be
+easy to learn. It is essential that the user understands the usage of the
+scripting language in our application before he/she is able to use it for
+analysis.
 
-\subsection{Reliability}
+\subsection{Code quality}
+In the project we strive to have good code readability and the code must be easy to understand. This is also an important part of the static analysis. We handle the pull based system on Github. For each new feature that will be implemented the author of that feature must create a new branch and do all his work in that branch so that the master branch always consists of a working version of the software. If someone wants to commit he has to make a pull request first. His code will then be reviewed by at least 2 teammembers and eventually fixed before it can be merged with the working version in the master branch.\\
+To improve code quality we have code conventions the writer should follow before
+his code is accepted. This is done by the Checkstyle plugin in the IDE that makes
```

Comment:
Klopt, maar wat ik probeer te zeggen is dat checkstyle alleen niet voldoende is. (Zoals vermeld in de planning feedback)

On 15 mei 2015, at 22:40, pjvanwijk <notifications@github.com<mailto:notifications@github.com>> wrote:


In docs/architecture/architecture.tex<https://github.com/clanghout/Health-Informatics-3/pull/51#discussion_r30444091>:

>
> -\subsection{Reliability}
> +\subsection{Code quality}
> +In the project we strive to have good code readability and the code must be easy to understand. This is also an important part of the static analysis. We handle the pull based system on Github. For each new feature that will be implemented the author of that feature must create a new branch and do all his work in that branch so that the master branch always consists of a working version of the software. If someone wants to commit he has to make a pull request first. His code will then be reviewed by at least 2 teammembers and eventually fixed before it can be merged with the working version in the master branch.\\
> +To improve code quality we have code conventions the writer should follow before
> +his code is accepted. This is done by the Checkstyle plugin in the IDE that makes


Er staan in de feedback een aantal tools zoals checkstyle waar we dus gebruik van moeten maken, dit lijkt me wel handig om te vermelden dat we dat dus ook gebruiken; aangezien dit iets is wat we zsm moeten gaan doen (kwam op mij over als verplicht)

Hier staat vermeld dat we checkstyle gebruiken ;)

—
Reply to this email directly or view it on GitHub<https://github.com/clanghout/Health-Informatics-3/pull/51/files#r30444091>.